drivers/spi/Kconfig.rpi_pico

```diff
@@ -9,3 +9,31 @@ config SPI_RPI_PICO_PIO
 	select PICOSDK_USE_CLAIM
 	help
 	  Enable driving SPI via PIO on the PICO
+
+if SPI_RPI_PICO_PIO
```

Comment:
These config items should be better to create as dts properties.

Reply:
The Kconfig options determine whether or not some good-sized code chunks are included when compiling spi_rpi_pico_pio.c.  Since half-duplex and word size options are set via the run-time operation parameter, they can't be defined in DTS.  I may be over-fussy about saving program memory here, and I'd appreciate any comments on whether this detailed level of control is appropriate.

Reply:
I'm concerned that pio-spi will be forced into a common configuration if there are instances on each of pio0 and pio1. There is no problem if these options are capabilities, but even in that case, I think they will be necessary to set them individually.

Reply:
I'm confused:  The Kconfig options don't control the configuration of the individual PIO peripherals;  that's done through DTS and the operation parameter.  The Kconfig options control what features are compiled into the driver.  If neither PIO requires half-duplex support, leaving CONFIG_SPI_RPI_PICO_PIO_HALF_DUPLEX unset will leave that code out of the compilation. If either PIO supports half-duplex, that option is necessary to make the half-duplex code available to whichever PIO needs it.  But it's still possible for one PIO to use a half-duplex configuration while the other uses the minimal 4-wire code.

Reply:
I think these configs can be replaced with dts and `DT_ANY_INST_HAS_PROP_STATUS_OKAY` macro.
We can write "if any spi-pio node has ... property, then ...." with this macro. It has the same meaning at the logical level.
By doing this, you should be able to respond more flexibly to the "hardware" configuration of spi-pio.

And these configs seem to define "hardware capabilities".
So, it is better to define in dts.
(I think that it is more appropriate to consider the functions configured by pio as "hardware" in the zephyr driver structure.)

> But it's still possible for one PIO to use a half-duplex configuration (by defining sio-gpios in DTS) while the other uses the minimal 4-wire code.

This isn't easy to understand from the description.

Reply:
@soburi - Turns out you were right:  the HALF_DUPLEX config option can, indeed, be replaced with DT_ANY_INST_HAS_PROP_STATUS_OKAY, based on detecting the presence of the sio-gpios in the DTS.  I've updated the code to do so.  Thanks for pursuing that issue!